fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/LakeTableTieringManager.java

```diff
@@ -0,0 +1,252 @@
+            // to avoid the existing tiering services are still tiering it
+            addLakeTable(tableInfo, TIERING_SERVICE_TIMEOUT_MS);
+        }
+    }
+
+    public void addLakeTable(TableInfo tableInfo) {
+        addLakeTable(tableInfo, null);
+    }
+
+    private void addLakeTable(TableInfo tableInfo, @Nullable Long delayMs) {
+        inLock(
+                lock,
+                () -> {
+                    tablePathById.put(tableInfo.getTableId(), tableInfo.getTablePath());
+                    tieringIntervalByTableId.put(
+                            tableInfo.getTableId(),
+                            tableInfo.getTableConfig().getDataLakeTieringInterval().toMillis());
+                    scheduleTableTiering(tableInfo.getTableId(), delayMs);
+                });
+    }
+
+    private void scheduleTableTiering(long tableId) {
+        scheduleTableTiering(tableId, null);
+    }
+
+    private void scheduleTableTiering(long tableId, @Nullable Long delayMs) {
+        Long tieringInterval = tieringIntervalByTableId.get(tableId);
+        if (tieringInterval == null) {
+            // the table has been dropped, return directly
+            return;
+        }
+        lakeTieringScheduleTimer.add(
+                new DelayedTiering(tableId, delayMs == null ? tieringInterval : delayMs));
+    }
+
+    public void removeLakeTable(long tableId) {
+        inLock(
+                lock,
+                () -> {
+                    tablePathById.remove(tableId);
+                    tieringIntervalByTableId.remove(tableId);
+                });
+    }
+
+    @Nullable
+    public Tuple2<Long, TablePath> requestTable() {
+        Long tableId = pendingTieringTables.poll();
+        // no any pending table, return directly
+        if (tableId == null) {
+            return null;
+        }
+        TablePath tablePath = tablePathById.get(tableId);
+        // the table has been dropped, request again
+        if (tablePath == null) {
+            return requestTable();
+        }
+        tieringLivenessProbe(tableId);
+        return new Tuple2<>(tableId, tablePath);
+    }
+
+    private void tieringLivenessProbe(long tableId) {
+        // we put the table to a timout timer,
+        // if no liveness for the table in this period of time, we consider the tiering is
+        // invalid and put it back to wait to be scheduled by other tiering service
+        lakeTieringScheduleTimer.add(
+                new TimerTask(TIERING_SERVICE_TIMEOUT_MS) {
+                    @Override
+                    public void run() {
+                        inLock(
+                                lock,
+                                () -> {
+                                    // the table is not in tiering, we put it back to wait to be
+                                    // scheduled
+                                    if (!liveTieringTableIds.remove(tableId)) {
+                                        scheduleTableTiering(tableId);
```

Comment:
should add to queue again? instead of waiting another interval?

Reply:
I agree it'll be better to put to queue instead of waiting another interval to make tier happen asap